src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1125,148 +1167,21 @@ export class SavedObjectsRepository {
-              _index: objectIndex,
-            },
-          ],
-        },
-      },
-      { ignore: [404] }
-    );
-    // exit early if a 404 isn't from elasticsearch
-    if (
-      isNotFoundFromUnsupportedServer({
-        statusCode: bulkGetResponse.statusCode,
-        headers: bulkGetResponse.headers,
-      })
-    ) {
-      throw SavedObjectsErrorHelpers.createGenericNotFoundEsUnavailableError(type, id);
-    }
-    const exactMatchDoc = bulkGetResponse?.body.docs[0];
-    const aliasMatchDoc = bulkGetResponse?.body.docs[1];
-    const foundExactMatch =
-      // @ts-expect-error MultiGetHit._source is optional
-      exactMatchDoc.found && this.rawDocExistsInNamespace(exactMatchDoc, namespace);
-    const foundAliasMatch =
-      // @ts-expect-error MultiGetHit._source is optional
-      aliasMatchDoc.found && this.rawDocExistsInNamespace(aliasMatchDoc, namespace);
-
-    let result: SavedObjectsResolveResponse<T> | null = null;
-    let outcomeStatString = REPOSITORY_RESOLVE_OUTCOME_STATS.NOT_FOUND;
-    if (foundExactMatch && foundAliasMatch) {
-      result = {
-        // @ts-expect-error MultiGetHit._source is optional
-        saved_object: getSavedObjectFromSource(this._registry, type, id, exactMatchDoc),
-        outcome: 'conflict',
-        alias_target_id: legacyUrlAlias.targetId,
-      };
-      outcomeStatString = REPOSITORY_RESOLVE_OUTCOME_STATS.CONFLICT;
-    } else if (foundExactMatch) {
-      result = {
-        // @ts-expect-error MultiGetHit._source is optional
-        saved_object: getSavedObjectFromSource(this._registry, type, id, exactMatchDoc),
-        outcome: 'exactMatch',
-      };
-      outcomeStatString = REPOSITORY_RESOLVE_OUTCOME_STATS.EXACT_MATCH;
-    } else if (foundAliasMatch) {
-      result = {
-        saved_object: getSavedObjectFromSource(
-          this._registry,
-          type,
-          legacyUrlAlias.targetId,
-          // @ts-expect-error MultiGetHit._source is optional
-          aliasMatchDoc
-        ),
-        outcome: 'aliasMatch',
-        alias_target_id: legacyUrlAlias.targetId,
-      };
-      outcomeStatString = REPOSITORY_RESOLVE_OUTCOME_STATS.ALIAS_MATCH;
-    }
-
-    await this.incrementResolveOutcomeStats(outcomeStatString);
-
-    if (result !== null) {
-      return result;
+    const { resolved_objects: bulkResults } = await internalBulkResolve<T>({
+      registry: this._registry,
+      allowedTypes: this._allowedTypes,
+      client: this.client,
+      serializer: this._serializer,
+      getIndexForType: this.getIndexForType.bind(this),
+      incrementCounterInternal: this.incrementCounterInternal.bind(this),
+      objects: [{ type, id }],
+      options,
+    });
+    const [result] = bulkResults;
+    if ((result as InternalBulkResolveError).error) {
+      throw (result as InternalBulkResolveError).error;
     }
-    throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
+    return result as SavedObjectsResolveResponse<T>;
```

Comment:
(Part 2 of 3)

In this section, the `SOR.resolve` method 1. calls `internalBulkResolve`, 2. checks to see if there was an error (throwing if need be), and 3. if there was no error, returns the result, casting it as a `SavedObjectsResolveResponse` type. This matches the existing behavior.